### DISCUSSION TITLE
Default number of threads for event loops

### DISCUSSION THREAD

**Discussion by andresmargalef** [netty/netty#13803] [General]
Hi everyone, maybe someone knows why the default value for the number of threads for event loop  [here](https://github.com/netty/netty/blob/6c883d620722c12e19561ce1a10b41fadfbd6a60/transport/src/main/java/io/netty/channel/MultithreadEventLoopGroup.java#L41) is `2 * number of logical cpu` instead of `number of logical cpu` because this value already takes account of hyperthreading.

**Comment by chrisvest** [comment]
Because there is still room for concurrency when some of them are blocked on IO or waiting for timers to trigger.

**Comment by andresmargalef** [comment]
Thanks!
